### ISSUE TITLE
Paging im REST-Service

### ISSUE THREAD

**Issue by Arne2** [xdoo/vaadin-demo#184]
Im MicroService soll Paging unterstützt werden.

**Comment by darenegade** [comment]
~~http://docs.spring.io/spring-data/rest/docs/1.1.x/reference/html/paging-chapter.html~~
http://docs.spring.io/spring-data/rest/docs/2.4.0.RELEASE/reference/html/#paging-and-sorting

**Comment by peter-mueller** [comment]
Bitte aktuelle Docs verwenden :arrow_up: .

**Comment by darenegade** [comment]
Ich schreib gleich eine Beschwerde an Google, dass sie die alten Docs höher anzeigen. :smile:

**Comment by darenegade** [comment]
TODO: Die Suche soll auch Paging unterstützen.

```


To use paging in your own query methods, you need to change the method 
signature to accept an additional Pageable parameter and return a Page 
rather than a List. For example, the following query method will be exported 
to /people/search/nameStartsWith and will support paging:
```

``` Java
@RestResource(path = "nameStartsWith", rel = "nameStartsWith")
public Page findByNameStartsWith(@Param("name") String name, Pageable p);
```

```
The Spring Data REST exporter will recognize the returned Page and give 
you the results in the body of the response, just as it would with a non-paged 
response, but additional links will be added to the resource to represent the 
previous and next pages of data.

```

Ansonsten wird man da bei Google schnell fündig :wink:

**Comment by FabianWilms** [comment]
Dickes Problem:
Bei Verwendung von paging wird bei allen Listen-Anfragen der Mandant umgangen. @PostFilter hat darauf keine Auswirkung.
Heißt: Man muss sich entscheiden, ob man lieber das Mandanten-Feature, oder das Paging haben möchte.

@darenegade @xdoo Was sollen wir da jetzt machen?

**Comment by xdoo** [comment]
@FabianHoltkoetter den Mandanten haben wir doch raus geworfen, oder? Dann sollte das kein Problem sein. Wichtig ist mir nur, dass auf der Oberfläche nicht Dinge von anderen Nutzern sichtbar sind, die der nicht frei gegeben hat.

@darenegade

**Comment by darenegade** [comment]
Stimmt, das Thema können wir wieder angehen. :+1:

**Comment by peter-mueller** [comment]
Ich glaube das das Problem mit dem paging der filter war, wenn man diesen also für die neuen Rechte implementiert könnte es sich wieder beißen

**Comment by FabianWilms** [comment]
Stimmt, das habe ich übersehen gehabt.

**Comment by xdoo** [comment]
Ja. Das muss man meiner Meinung aber nur dokumentieren. Ohne Filter - und das ist ja der Normalfall - geht es.
